flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/dataSet/DataSetAggregateRule.scala

```diff
@@ -50,6 +51,13 @@ class DataSetAggregateRule
       agg.getGroupSets.size() == 1) {
       return false
     }
+    // check if we have over aggregates
+    val overAggNames =
```

Comment:
I thought about this again. 
The optimization rules are not the right place for this check and should not be modified. 

Instead the check should be moved to the Table API validation phase. 
SQL queries are automatically checked by Calcite's `SqlValidator`.